OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -266,11 +272,16 @@ public virtual bool TickUnload(Actor self, Actor proc)
 						contents.Remove(resourceType);
 
 					currentUnloadTicks = Info.BaleUnloadDelay;
+					currentUnloadFinishedDelayTicks = Info.UnloadFinishedDelay;
 					UpdateCondition(self);
 					return false;
 				}
 			}
 
+			// Wait until artificial delay time has passed
+			if (--currentUnloadFinishedDelayTicks > 0)
+				return false;
+
```

Comment:
This isn't the right way to implement it. You should be using activities, see from where the TickUnload function is called. The proper way would be using `QueueChild(new Wait(UnloadFinishedDelay));` on the loop ending from `HarvesterDockSequence`

I could be a bit wrong on the specifics, I haven't needed to implement anything like this

Reply:
Though actually that might be wrong as well. Your implementation of `currentUnloadFinishedDelayTicks` handling could just be moved to  `HarvesterDockSequence` and I think it would be fine

Reply:
Oh, alright. I was copying the functionality already present in-code there. I'll look into moving it into a more conventional place, thanks!